### PR TITLE
[provider-ui] - Add initial support for Cypress

### PR DESCRIPTION
**Pull request by tsingh2k15**
- Add skeleton structure for Cypress test automation
- Add canary test which always pass!

**Description**

This PR contributes towards https://github.com/layer5io/meshery/issues/1037 

**Notes to reviewers**

#### Checklist

- [x] [Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin).
- [x] PR title includes `[<name of area>]:<description>`

### CODE REVIEW

File: provider-ui/package.json [hunk 1]
```diff
@@ -27,9 +27,10 @@
     "styled-jsx": "latest"
   },
   "devDependencies": {
+    "babel-eslint": "^10.1.0",
+    "cypress": "^4.8.0",
```
Comment (kushthedude) [inline provider-ui/package.json:31]: Can you please incorporate the provider-ui cypress tests in Github-Actions as done for meshery-ui ?

File: provider-ui/package.json [hunk 2]
```diff
@@ -27,9 +27,10 @@
     "styled-jsx": "latest"
```
Comment (kushthedude) [inline provider-ui/package.json:27]: Can you edit the `npm test` script in `provider-ui/package.json` to run cypress with it?

File: provider-ui/.eslintignore [hunk 3]
```diff
@@ -4,5 +4,6 @@ out
 static
```
Comment (kushthedude) [inline provider-ui/.eslintignore:4]: Can you add the cypress command in `makefile` too as done for meshery-ui ?

**Comment by tsingh2k15** [comment]
> Hey @tsingh2k15 , great work here it was indeed another step while introducing tests for the UI.
> A few suggestions if you will look into it!

sure @kushthedude , i will incorporate the review comments. Thanks!

File: provider-ui/.eslintignore (hunk 3 above)
Comment (tsingh2k15) [reply provider-ui/.eslintignore:4]: @kushthedude sorry, i could not understand this one, can you clarify?

Comment (kushthedude) [reply provider-ui/.eslintignore:4]: `make run-provider-ui-test` to execute command `cd provider-ui; npm test; cd ..`

**Comment by leecalcote** [comment]
@tsingh2k15 we have a couple conflicts that you might take a look at in the: ```provider-ui/package-lock.json
provider-ui/package.json```

**Comment by leecalcote** [comment]
I went ahead and resolved the outstanding merge conflicts.

**Comment by tsingh2k15** [comment]
> There are a couple conflicts that @tsingh2k15 will want to take a look at.

resolved

**Comment by tsingh2k15** [comment]
> > Hey @tsingh2k15 , great work here it was indeed another step while introducing tests for the UI.
> > A few suggestions if you will look into it!
> 
> sure @kushthedude , i will incorporate the review comments. Thanks!

@kushthedude updated PR with requested changes, please review when you get chance. Thanks!

**Comment by tsingh2k15** [comment]
@leecalcote  This is ready for your review. Thanks!

**Comment by kushthedude** [comment]
Great Work @tsingh2k15 ! 🥳

**Comment by tsingh2k15** [comment]
thank you! I will start looking into writing the tests.

**Comment by leecalcote** [comment]
Sweet, sweet, sweet. @tsingh2k15 👍

**Pull request merged**